### ISSUE TITLE
Develop Embedding Layers

### ISSUE THREAD

**Issue by RishabhMalviya** [RishabhMalviya/raag-midi-gen#1]
# Completion Criteria
You should have a functioning python class (sub-classed from `torch.nn.Module`) that takes in an element from the MIDI `Dataset`, and outputs a torch tensor of shape `(len(melody), embedding_size)`.

# Details
## Embeddings for Note On events
Concatenate multiple embeddings for each Note On event:

1. Note (C, C#, D, etc)
2. Octave
3. Velocity
4. Duration
5. Positional Embeddings (capture variation across time scales)
    - sin(w_beat * t)
    - cos(w_beat * t)
    - sin(w_measure * t)
    - cos(w_measure * t)
    - sin(w_melody * t)
    - cos(w_melody * t)

## Find a way to inject embeddings for special tokens
Then you need embeddings for each of these special tokens too:
1. BOS
2. EOS
3. PAD (hard set it to all zeros)
4. MASK
5. EMPTY

**Comment by RishabhMalviya** [comment]
Don't encode the duration. Just encode NoteOn and NoteOff events separately.
--- OR ---
Encode the duration as a fraction of the total melody length.


^Using the second option for now
